calibrators/mapping_based_calibrator/README.md

```diff
@@ -0,0 +1,238 @@
+# mapping_based_calibrator
+
+A tutorial for this calibrator can be found [here](../docs/tutorials/mapping_based_calibrator.md)
+
+## Purpose
+
+The package `mapping_based_calibrator` allows extrinsic calibration between 3d lidar and 3d lidar sensors, as well as baselink and 3d lidar sensors used in autonomous driving and robotics.
+
+## Inner-workings / Algorithms
+
+### lidar-lidar calibration
+
+The calibrator is designed to estimate the transformation between lidar and lidar sensors. It starts by employing a lidar, called the `mapping lidar`, for mapping purposes. Then, it utilizes pointcloud registration algorithms to calibrate the remaining lidars, known as `calibration lidars`, to the `mapping lidar`. Specifically, the calibration process encompasses three primary steps: constructing a map, preparing calibration data, and finally estimating the transformations.
+
+A prerequisite for this calibrator is an environment rich in natural landmarks suitable for registration-based mapping in all directions, ensuring that the lidar captures sufficient details beyond simple features like lane surfaces or walls.
+
+#### Step 1: Construct a map
+
+The calibrator designates one of the lidars (defined in the launch file) as the `mapping lidar` for mapping purposes. Pointclouds from this lidar utilize either the NDT or GICP algorithm to calculate the pose and also store the pointclouds as a map for future usage. The following rules are applied while constructing the map.
+
+- For each `N` meters, a keyframe will be saved, which is used for mapping and estimating transformation.
```

Comment:
A new concept was used but nothing has been explained.
Concepts like Frames and Keyframes **(this is not an exhaustive list)**, need to be explained before using them